[DOCS] Clarify field data cache behavior (#64375)
elastic/elasticsearch: docs/reference/modules/indices/fielddata.asciidoc

[[modules-fielddata]]
=== Field data cache settings

The field data cache contains <<fielddata-mapping-param, field data>> and <<eager-global-ordinals, global ordinals>>,
which are both used to support aggregations on certain field types.
Since these are on-heap data structures, it is important to monitor the cache's use.

[discrete]
[[fielddata-sizing]]
==== Cache size

The entries in the cache are expensive to build, so the default behavior is
to keep the cache loaded in memory. The default cache size is unlimited,
causing the cache to grow until it reaches the limit set by the <<fielddata-circuit-breaker, field data circuit breaker>>. This behavior can be configured.

If the cache size limit is set, the cache will begin clearing the least-recently-updated
entries in the cache. This setting can automatically avoid the circuit breaker limit,
at the cost of rebuilding the cache as needed.

If the circuit breaker limit is reached, further requests that increase the cache

> size will be prevented. In this case you shoul manually <<indices-clearcache, clear the cache>>.

Reviewer: shoul -> should

`indices.fielddata.cache.size`::
(<<static-cluster-setting,Static>>)
The max size of the field data cache, eg `38%` of node heap space, or an

> absolute value, eg `12GB`. Defaults to unbounded. Should be set smaller than the
> <<fielddata-circuit-breaker>>, if set.

Reviewer 2: It's not clear whether `if set` applies to the circuit breaker or this setting.
```suggestion
absolute value, eg `12GB`. Defaults to unbounded. This value should not
exceed the <<fielddata-circuit-breaker,`indices.breaker.fielddata.limit`>>.
```

> <<fielddata-circuit-breaker>>, if set.

Reviewer 3: As @jtibshirani mentioned, I'd avoid recommending size-based eviction for this cache.
